app/services/katello/candlepin/server.rb

```diff
@@ -0,0 +1,20 @@
+module Katello
+  module Candlepin
+    class Server
+      def self.init
+        return unless SETTINGS[:katello][:use_candlepin] && !Foreman.in_rake?('db:migrate')
+        candlepin_status = Katello::Ping.ping([:candlepin])
+
+        if candlepin_status[:status] == Katello::Ping::FAIL_RETURN_CODE
+          fail _("There was an issue with the Candlepin service %s: ") % candlepin_status[:candlepin][:message]
+        end
+
+        Organization.all.each do |org|
+          User.current = User.anonymous_admin
+          ForemanTasks.sync_task(::Actions::Katello::Organization::Create, org) unless org.library
```

Comment:
Oh you're right and I could have seen that if I read it. That still leaves the option for a pre check avoiding a ping to an external service.